python/ray/_private/function_manager.py

```diff
@@ -417,7 +417,6 @@ def _wait_for_function(self, function_descriptor, job_id, timeout=10):
             # Try importing in case the worker did not get notified, or the
             # importer thread did not run.
             self._worker.import_thread._do_importing()
-            time.sleep(0.001)
```

Comment:
why sleep is removed here? What if there actually is a function lost, will this eventually slowdown the GCS?

Reply:
If a function is lost, the cluster would need to be restarted anyway. Otherwise, there should be no additional load. If we are really worried about the load, we can add a backoff on the order of 1s.

Reply:
I don't think we need to restart the whole cluster. It's per-job, so we can just stop the job and run a new one.

But if we don't do the backoff, my concern is that it's going to affect the other jobs.

Reply:
The previous backoff of 1ms shouldn't make a big difference. Changed the retry loop to at most 20s, with at least 1s backoff.